kernel/sched.c

```diff
@@ -1055,6 +1055,13 @@ void z_sched_abort(struct k_thread *thread)
 		return;
 	}
 
+	/* if the thread is already terminated in other
+	 * cores, e.g. in thread exit, just return
+	 */
+	if (thread->base.thread_state & _THREAD_DEAD) {
+		return;
+	}
```

Comment:
Is this really needed?  There is a loop checking for _THREAD_DEAD immediately after the IPI that follows this line, which should be guaranteed safe (the target thread obviously isn't running, it's dead).

Basically, this sounds like a pure optimization to me where we can avoid an IPI in cases where user code is trying to abort a thread that already exited.  Is that really a case worth optimizing?

If it's a bug fix instead, then we should document that.

Reply:
This part will in a separete PR to trace. It's more like an optimization to avoid unnecessary ipi interrupt.  I am also thinking whether multiple abort ops on the thread will have side effects, especially in SMP.